core/src/blocktree_processor.rs

```diff
@@ -27,34 +30,37 @@ fn first_err(results: &[Result<()>]) -> Result<()> {
 fn par_execute_entries(
     bank: &Bank,
     entries: &[(&Entry, LockedAccountsResults<Transaction>)],
+    thread_pool: &ThreadPool,
```

Comment:
now that I see what this costs in terms of code churn and test-ability, how about:

https://doc.rust-lang.org/std/macro.thread_local.html

Reply:
doing it this way would also help with sigverify...

Reply:
To be on the same page, are you suggesting to use thread local to store the thread_pool?

Reply:
yeah, exactly

Reply:
isn't that going to be way worse at the end of the day?  why not just create a pool and destroy it every time?  I am not sure perf will be that bad.

Reply:
wondering why would it be worse... The replay stage runs one thread that's calling into blocktree processor. So instead of creating threadpool on the stack of thread, it's storing it in TLS.

Reply:
i mean API/programing wise.  TLS is this weird global that is not really a global.

Reply:
Got it...

Reply:
prior to this, we were using a global that was really a global and getting bit